dynamiq/nodes/tools/exa_search.py

```diff
@@ -115,23 +136,37 @@ def _format_search_results(self, results: list[dict[str, Any]]) -> str:
     def execute(self, input_data: ExaInputSchema, config: RunnableConfig | None = None, **kwargs) -> dict[str, Any]:
         """
         Executes the search using the Exa API and returns the formatted results.
+
+        Input parameters override node parameters when provided.
         """
         logger.info(f"Tool {self.name} - {self.id}: started with input:\n{input_data.model_dump()}")
 
         config = ensure_config(config)
         self.run_on_node_execute_run(config.callbacks, **kwargs)
 
-        payload = {k: v for k, v in input_data.model_dump().items() if v is not None}
-        include_full_content = payload.pop("include_full_content", False)
-
         payload = {
-            "query": payload.pop("query"),
-            "useAutoprompt": payload.pop("use_autoprompt", False),
-            "type": payload.pop("query_type", "neural"),
-            "numResults": payload.pop("limit", 10),
-            **{self.to_camel_case(k): v for k, v in payload.items()},
+            "query": input_data.query,
+            "useAutoprompt": input_data.use_autoprompt or self.use_autoprompt,
+            "type": input_data.query_type or self.query_type,
+            "numResults": input_data.limit or self.limit,
+            "includeDomains": input_data.include_domains or self.include_domains,
+            "excludeDomains": input_data.exclude_domains or self.exclude_domains,
+            "includeText": input_data.include_text or self.include_text,
+            "excludeText": input_data.exclude_text or self.exclude_text,
+            "category": input_data.category or self.category,
         }
 
+        input_dict = input_data.model_dump(exclude_unset=True)
+        include_full_content = self.include_full_content
+
+        if "include_full_content" in input_dict:
```

Comment:
Why we here using another logic if `include_full_content` mandatory param for node? (make this param in schema optional as you made for the rest )